simplified-accessibility/src/main/java/org/nypl/simplified/accessibility/AccessibilityToasts.kt

```diff
@@ -0,0 +1,17 @@
+package org.nypl.simplified.accessibility
+
+import android.content.Context
+import android.widget.Toast
+
+/**
+ * The default implementation of the [AccessibilityToastsType] interface that just displays
+ * a toast using the given context.
+ */
+
+class AccessibilityToasts(
+  private val context: Context
+) : AccessibilityToastsType {
+  override fun show(message: String) {
+    Toast.makeText(this.context, message, Toast.LENGTH_LONG).show()
```

Comment:
You might see some glitchy behavior here when multiple messages are shown in rapid succession.

Best practice is to retain the Toast object that's created and update it instead of creating a new one for each message. You'll want to do something like this:

```kotlin
  this.toast = this.toast ?: Toast(context).apply { setDuration(Toast.LENGTH_LONG) }
  this.toast.setText(message)
  if (!this.toast.view.isShown()) {
    this.toast.show()
  }
```

Reply:
Is this just a visual issue, or would it affect the screen reader too? Remember the sole purpose of these is _not_ to be seen. :laughing:

Reply:
Well, if you don't want this to actually be visible then I suggest sending an accessibility event to announce instead.

You'll need a view reference to do this though.

```kotlin
view.announceForAccessibility(message)
```

> Source: https://developer.android.com/reference/android/view/View#announceForAccessibility(java.lang.CharSequence)

This is just shorthand for something like:

```kotlin
val manager = context.getSystemService(Context.ACCESSIBILITY_SERVICE) as AccessibilityManager

if (manager.isEnabled()) {
    AccessibilityEvent e = AccessibilityEvent.obtain().apply {
        setEventType(AccessibilityEvent.TYPE_ANNOUNCEMENT)
        setClassName(context.getClass().getName())
        setPackageName(context.getPackageName())
        getText().add(message)
    }
    manager.sendAccessibilityEvent(e)
}
```

I haven't tested this myself so you'll want to verify it works properly.

Reply:
:weary: 

I sat there for about half an hour looking for a way to explicitly "announce" messages to the screenreader. _Of course_ it's defined on `View`. Of course.

Won't use toasts at all for this now.

Reply:
Hah! Yeah, it's not super well documented. I think the accessibility team at Google was only a couple people at one point. I only know cause I've had to do this once before :P